crypto/http/http_lib.c

```diff
@@ -21,11 +21,12 @@
  */
 
 int OSSL_HTTP_parse_url(const char *url, char **phost, char **pport,
-                        char **ppath, int *pssl)
+                        int *pport_num, char **ppath, int *pssl)
 {
     char *p, *buf;
-    char *host;
-    const char *port = OSSL_HTTP_PORT;
+    char *host, *host_end;
+    const char *path, *port = OSSL_HTTP_PORT;
+    long portnum = 80;
     size_t https_len = strlen(OSSL_HTTPS_NAME);
 
     if (!ossl_assert(https_len >= strlen(OSSL_HTTP_NAME)))
```

Comment:
This is a very strange check..

Reply:
I've added a comment explaining it:
```
/* https_len is used below as an upper bound of the prefix length */
```

Reply:
My point was that it is comparing the length of 2 fixed strings.. Unless someone plans on redefining these strings this seems like a strange thing to check.

Reply:
Yeah, but meanwhile I found that the whole `https_len` business is not needed any more 
after I strengthened the scheme parsing.